File: app/src/main/java/com/footinit/jetpack/NotificationWorker.java
package com.footinit.jetpack;


import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Build;

import java.util.Random;

import androidx.annotation.NonNull;
import androidx.core.app.NotificationCompat;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class NotificationWorker extends Worker {

    public static final String TAG = NotificationWorker.class.getSimpleName();

    public NotificationWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        displayNotification();
        return Result.success();
    }

    private void displayNotification() {
        NotificationManager notificationManager = (NotificationManager) getApplicationContext()
                .getSystemService(Context.NOTIFICATION_SERVICE);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(TAG, TAG, NotificationManager.IMPORTANCE_DEFAULT);
            notificationManager.createNotificationChannel(channel);
        }

        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext(), TAG)
                .setContentTitle("WorkManager")
                .setContentText("Periodic Request" + new Random().nextInt(500))
                .setSmallIcon(R.mipmap.ic_launcher);

        notificationManager.notify(new Random().nextInt(1000), builder.build());
    }
}

File: app/src/main/java/com/footinit/jetpack/NotificationWorker.kt
package com.footinit.jetpack

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build
import androidx.core.app.NotificationCompat
import androidx.work.Worker
import androidx.work.WorkerParameters
import java.util.*

class NotificationWorker(context: Context, workerParams: WorkerParameters) :
        Worker(context, workerParams) {

    companion object {
        val TAG = NotificationWorker::class.java.simpleName
    }

    override fun doWork(): Result {
        displayNotification()
        return Result.success()
    }

    private fun displayNotification() {
        val notificationManager = applicationContext
                .getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(TAG, TAG, NotificationManager.IMPORTANCE_DEFAULT)
            notificationManager.createNotificationChannel(channel)
        }

        val builder = NotificationCompat.Builder(applicationContext, TAG)
                .setContentTitle("WorkManager")
                .setContentText("Periodic Request" + Random().nextInt(500))
                .setSmallIcon(R.mipmap.ic_launcher)

        notificationManager.notify(Random().nextInt(1000), builder.build())
    }
}